### ISSUE TITLE
Market history plugin: `history-per-size` option doesn't work

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-core#472]
Old data never got removed.

**Comment by oxarbitrage** [comment]
reproduced, it seems to be a problem around here: https://github.com/bitshares/bitshares-core/blob/master/libraries/plugins/market_history/market_history_plugin.cpp#L129 and https://github.com/bitshares/bitshares-core/blob/master/libraries/plugins/market_history/market_history_plugin.cpp#L202

ill check it more in deep tomorrow and try to fix it.

**Comment by abitmore** [comment]
@oxarbitrage I'll handle this. Thank you anyway.

**Comment by abitmore** [comment]
Test result with current BitShares main net (head block `#21742070`) and following options:
```
# plugins =
# track-account =
partial-operations = 1
max-ops-per-account = 1000
bucket-size = [60,300,900,1800,3600,14400,86400]
history-per-size = 1000
```
running in Ubuntu 16.04,
before fix: `VIRT = 6829564`, `RES = 5.272g`
after fix: `VIRT = 5982640`, `RES = 4.805g`

So the fix saves 470M ~ 850M RAM.

**Comment by oxarbitrage** [comment]
fixed at https://github.com/bitshares/bitshares-core/pull/478
